### PR TITLE
Undefined Behaviour in C/CPP

### PR DESCRIPTION
**Pull request by elixir14082002**
<hr>

## Description 📜

Added a documentation on Undefined Behaviour in C/CPP.
#7424

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by elixir14082002** [comment]
@Suvraneel Please review my PR!

File: C_CPP/Fundamentals/Undefined Behaviour/Undefined-Behaviour.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+### When does an *undefined behaviour* occur?
+Undefined behaviour occurs due to the following reasons:
+- Dividing a number by zero.
+- Trying to access a null pointer.
+- Using uninitialized variables.
+- Going out of bounds.
+- Going beyond the limit of the datatype.
+
+The above are the few common mistakes which lead to undefined behaviour. There are also other reasons that that lead to this kind of behaviour.
```
Comment (vasu-1) [inline C_CPP/Fundamentals/Undefined Behaviour/Undefined-Behaviour.md:14]:
```suggestion
The above are the few common mistakes which lead to undefined behaviour. There are also other reasons that lead to this kind of behaviour.
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]

**Comment by elixir14082002** [comment]
@vasu-1 @Suvraneel I made the required changes.

**Review by vasu-1 — CHANGES_REQUESTED** [review]
.

**Review by vasu-1 — APPROVED** [review]
LGMT @elixir14082002 

@Saviour1001 Please Check it for Once !

**Pull request merged**